### PR TITLE
throttle implemented

### PR DESCRIPTION
**Pull request by island205**

### CODE REVIEW

File: test/unit/throttle.spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }, 400);
+
+    // should still be 0 because trailing call a canceled
+    setTimeout(() => {
+      editor.counter.should.equal(0);
+      done();
+    }, 600);
+  });
+
+  it('separate instances do not share timers', function (done) {
```
Comment (jayphelps) [inline test/unit/throttle.spec.js]: test titles need to be a sentence structured to make sense with "it" at the beginning (the name of the function).

e.g.

```js
it('does not share timers between instances', function (done) {
```

Same issue above,

```js
it('does not invoke function as "leading" or "trailing" if options are false', function (done) {
```

**Comment by jayphelps** [comment]
Awesome! I personally prefer the underscore/lodash way of providing extraneous options like leading/trailing/etc as a hash. This allows us to continue to add new ones and without confusion about the expected order. Thoughts?

```js
class Foo {
  @throttle(500, { leading: true, trailing: true })
  updateContent(content) {}
```

**Comment by island205** [comment]
@jayphelps done.

**Comment by jayphelps** [comment]
Thanks @island205! :tada:

**Comment by tleunen** [comment]
Wasn't possible to just call `lodash.throttle`?

**Comment by jayphelps** [comment]
@tleunen if you're using classes, not by itself. You could certainly use our `@​decorate` decorator though,  previously.

```js
import { decorate } from 'core-decorators';
import { throttle } from 'lodash';

class Foo {
  @decorate(throttle, 500)
  updateContent(content) { }
}
```
Ultimately, I still wanted throttle to be part of core, though won't be adding the other lodash/underscore helpers at this time.

**Pull request merged**